lib/practice_exercises.rb

```diff
@@ -1,13 +1,58 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) one loop based on an array length.
+# Space Complexity: O(n) 3n but we drop the constant. I made 3 new arrays from one string.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+
+  concate_array = string.downcase.split("")
+  backwards = ""
+  concate_array_no_punc = ""
+
+  concate_array.length.times do |i|
+    if [":",","," "].include?(concate_array[-1-i])
+      nil 
+    else  
+      backwards += concate_array[-1-i]
+    end
+
+    if [":",","," "].include?(concate_array[i])
+      nil 
+    else  
+      concate_array_no_punc += concate_array[i]
+    end
+  end
+
+  if backwards == concate_array_no_punc
+    return true
+  else 
+    return false
+  end
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) two nested arrays each dependent on the size of the array
+# Space Complexity: O(1) I only deal with the one array. 
+
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍